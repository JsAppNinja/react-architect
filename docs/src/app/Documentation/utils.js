export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
