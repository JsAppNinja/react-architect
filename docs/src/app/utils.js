import React from 'react';
export const githubHref = 'https://github.com/mlaursen/react-md';

const lorems = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at placerat nulla. Morbi eu erat diam. Donec malesuada mauris in neque efficitur luctus. Vestibulum porttitor elit velit, pellentesque imperdiet tellus maximus in. Phasellus consectetur lacus nunc, quis efficitur felis feugiat ac. Vivamus non erat laoreet, volutpat tellus sit amet, porttitor orci. Sed scelerisque id arcu at ornare. Suspendisse potenti. Phasellus eu posuere mi. Fusce venenatis odio dui, ac accumsan lacus dapibus et. Praesent euismod nulla at porta elementum. Nunc ullamcorper velit aliquam velit faucibus tempus.',
  'Duis ligula elit, elementum a nisl a, venenatis fermentum libero. Cras bibendum dignissim nisl, sed sodales tellus scelerisque sed. Sed at enim semper, imperdiet elit et, venenatis magna. Morbi vestibulum purus in ante placerat viverra. Nullam molestie arcu ut dolor rutrum varius. Sed vehicula laoreet volutpat. Quisque feugiat, nisl in sollicitudin consectetur, mi metus aliquet urna, ac sagittis nulla nulla in diam. Fusce id enim auctor ante fringilla auctor in et quam.',
  'Donec pulvinar ligula augue, et tincidunt velit bibendum non. Aenean id metus nunc. Donec non feugiat nisl. In accumsan ligula purus, ut vulputate elit tempus sit amet. Nullam volutpat semper lectus. Nunc a libero tellus. Duis posuere nisi diam, nec euismod risus interdum sit amet. Pellentesque at lacus bibendum, cursus ante non, maximus arcu. Morbi id leo nulla. Integer quis feugiat quam. Etiam auctor pharetra vestibulum.',
  'Nulla sodales metus metus. Sed a metus quis ipsum sagittis placerat. Nunc sit amet molestie purus, nec rutrum quam. Praesent elementum risus eget mattis condimentum. Morbi dignissim, dolor et finibus sollicitudin, nulla enim molestie est, eget tincidunt magna libero et dolor. Fusce tempor sapien et urna sodales, at laoreet nunc egestas. Aenean eget arcu in felis ultrices vehicula. Maecenas interdum viverra sem vitae sodales. Duis tincidunt varius nunc sed viverra. Duis sit amet semper ex, ultrices aliquet mi. Donec a congue sem, ac facilisis ex.',
  'Proin venenatis tortor et ex viverra, ac facilisis mi ullamcorper. Aenean eget est bibendum, fermentum enim eget, semper orci. Nunc magna mauris, fermentum sit amet diam ut, semper malesuada justo. Fusce faucibus tellus eget arcu imperdiet, aliquet placerat massa molestie. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Maecenas gravida quam id ultricies accumsan. Sed porta accumsan libero sed venenatis. Morbi scelerisque, ex in gravida porttitor, ipsum erat sodales urna, mattis mattis nibh ligula quis tellus. Pellentesque consequat et metus scelerisque dapibus. Praesent at maximus ligula. Phasellus vel orci tortor.',
  'Mauris posuere nisl sit amet justo condimentum volutpat. Aliquam venenatis rhoncus est, tristique semper tortor iaculis et. Donec pretium metus nec sodales mollis. Ut vel ipsum sapien. Fusce sit amet libero efficitur orci aliquet scelerisque a ut arcu. In hac habitasse platea dictumst. Sed turpis libero, dictum nec imperdiet nec, maximus eget urna. Sed vel lorem sit amet nisi vulputate mattis. In in odio erat. Aenean euismod lorem nec viverra auctor. Mauris id ipsum maximus, tempus dui nec, mollis felis. Etiam sodales metus sit amet lacus aliquet tristique. Nulla in lacus massa. Morbi quis lorem vitae dolor maximus luctus vel vel eros. Phasellus posuere in purus eget convallis.',
  'Aliquam mauris justo, ornare quis quam fermentum, mollis volutpat dolor. Donec sagittis velit non nisl porttitor, id lacinia tortor vulputate. Suspendisse posuere at felis eu fermentum. Donec at vehicula leo, id sollicitudin ipsum. Donec dapibus sodales erat at malesuada. Mauris sollicitudin velit sed arcu molestie congue. Suspendisse ac arcu enim. Sed ac urna id mi rutrum ornare sit amet in nisl. Phasellus at tempus magna. Nulla facilisi. Mauris pharetra finibus mi ut facilisis. Duis mattis odio id quam vestibulum, id consequat ex fringilla. Aenean sollicitudin accumsan sollicitudin. Duis euismod maximus tempor. Phasellus mi enim, bibendum at nisi vel, cursus venenatis sem. Praesent sagittis, nunc at rhoncus maximus, ligula turpis imperdiet eros, pharetra tincidunt tortor magna id sem.',
  'Sed eleifend sapien condimentum iaculis cursus. In vel pretium turpis. Duis molestie euismod turpis id consectetur. Nunc pharetra sapien libero, a volutpat erat finibus eget. Duis euismod eleifend felis consequat fringilla. Sed varius diam ligula, quis rutrum ex mollis non. Phasellus ac malesuada nibh. Praesent sit amet auctor urna. Quisque eget magna id ex dapibus vulputate eget vel magna.',
  'Ut viverra tempor placerat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras neque diam, auctor eget sapien et, tincidunt blandit turpis. Nam non volutpat velit. Aliquam iaculis euismod nisl non facilisis. Ut blandit dignissim turpis in finibus. Cras neque augue, tempus non neque eu, ornare ultrices sem. Nullam ultricies mi quis semper pellentesque. Mauris mauris lacus, ullamcorper a iaculis quis, tempus id tellus. Phasellus in orci orci. Fusce consequat suscipit ipsum ac bibendum. Maecenas et leo eu leo iaculis varius. Nam consectetur laoreet nisl blandit condimentum.',
  'Ut dictum, sapien vel malesuada posuere, tortor tellus blandit risus, eget hendrerit quam diam eu nibh. Aliquam erat volutpat. Ut rhoncus augue ac nunc porttitor hendrerit. Pellentesque enim risus, auctor quis turpis eget, pulvinar malesuada urna. Nullam tortor mi, elementum id sodales eget, aliquam et nulla. Pellentesque efficitur mi eros, in rutrum metus ultrices vel. Proin vitae diam sit amet nunc pretium malesuada. Fusce at nisi scelerisque, ultrices tortor nec, finibus velit. Donec molestie augue eget leo rutrum, nec ornare arcu gravida. Quisque tincidunt risus elit, ut condimentum eros egestas congue.',
];
export function loremIpsum(paras = 3) {
  let ipsums = [];
  for(let i = 0; i < paras; i++) {
    ipsums.push((
      <p key={i}>{lorems[Math.floor(Math.random() * lorems.length)]}</p>
    ));
  }
  return ipsums;
}

export const numstr = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

export const randomImage = ({ width, height, time } = {}) => {
  width = !width ? 200 : width;
  height = !height ? 200 : height;
  time = typeof time === 'undefined' ? Date.now() : time;
  return `https://unsplash.it/${width}/${height}/?random&time=${time}`;
};

export const hostPrefix = ''; // change to '/react-md' for production
export const imgPrefix = hostPrefix + '/imgs';
