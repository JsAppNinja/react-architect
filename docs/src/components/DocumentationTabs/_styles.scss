@import '~globals';

.md-title--toolbar {
  ~ .documentation-tabs {
    left: 0;
  }
}

.md-title--permanent-offset,
.md-title--persistent-offset {
  ~ .documentation-tabs {
    left: $md-drawer-tablet-width;
  }
}

.md-tabs.documentation-tabs {
  bottom: 0;
  position: absolute;
  right: 0;
}
