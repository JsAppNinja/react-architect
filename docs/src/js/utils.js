import React from 'react';

import { FontIcon, Avatar } from 'react-md';

export const githubHref = 'https://github.com/mlaursen/react-md';

export const mainLinks = [{
  link: '',
  primaryText: 'Home',
  leftIcon: <FontIcon>home</FontIcon>,
}, {
  link: 'sassdoc',
  primaryText: 'SASS Doc',
  leftAvatar: <Avatar src="/imgs/sass-icon.png" alt="SASS Icon" />,
}];
