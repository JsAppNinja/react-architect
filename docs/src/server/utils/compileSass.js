const path = require('path');
const Promise = require('bluebird');
const nodeSass = require('node-sass');

const sass = Promise.promisify(nodeSass.render);

const { ACCENTABLE_COLORS, PRIMARY_COLORS, SECONDARY_HUES } = require('../../shared/constants/colors');

const DARK_THEME_STYLES = `
.example-code,
.code-block {
  background: $md-grey-300;
}

a:not(.md-btn) {
  color: $md-purple-a-400;

  &:visited {
    color: $md-purple-a-100;
  }
}
`;

function toColorName(color) {
  return color && color.replace('_', '-');
}

function validatePrimary(color) {
  return color && PRIMARY_COLORS.indexOf(color) !== -1;
}

function validateSecondary(color, hue) {
  return color && hue && ACCENTABLE_COLORS.indexOf(color) !== -1 && SECONDARY_HUES.indexOf(hue) !== -1;
}

/**
 * Attempts to compiles sass based off of a css file name. The filename should be formatted
 * as 'PRIMARY_COLOR-SECONDARY_COLOR-HUE-dark?.css' where `dark` is optional.
 *
 * @param {String} name - The CSS filename to convert into styles.
 * @return {Promise} a promise with the buffer of compiles styles
 */
module.exports = function compileSass(name) {
  const [p, s, hue, dark] = name.split('-');
  const primary = toColorName(p);
  const secondary = toColorName(s);

  if (!validatePrimary(primary)) {
    return Promise.reject(new Error(`Invalid primary color: ${primary}`));
  } else if (!validateSecondary(secondary, parseInt(hue, 10))) {
    return Promise.reject(new Error(`Invalid secondary color: ${secondary}-${hue}`));
  }

  const primaryColor = `$md-${primary}-500`;
  const secondaryColor = `$md-${secondary}-a-${hue}`;
  const darkThemeStyles = dark ? DARK_THEME_STYLES : '';

  const styles = `
@import 'react-md';

$md-primary-color: $md-light-blue-500;
$md-secondary-color: $md-deep-orange-a-200;
@include react-md-theme-everything(${primaryColor}, ${secondaryColor}, ${!dark}, 'custom-theme');

.custom-theme {
  .banner {
    background: ${primaryColor};
  }

  .react-md-logo {
    text,
    #Group > path:first-child,
    path#Oval-210 {
      fill: ${secondaryColor};
    }
  }

  ${darkThemeStyles}
}

@media #{$md-tablet-media} {
  .custom-theme .phone-size-container .md-toolbar:not(.md-toolbar--inset)::before {
    background: ${primaryColor};
  }
}
  `;

  return sass({
    data: styles,
    includePaths: [path.resolve(process.cwd(), '..', 'src', 'scss')],
    outputStyle: 'compressed',
  }).then((stats, error) => {
    if (error) {
      return Promise.reject(error);
    }

    return stats.css;
  });
};
