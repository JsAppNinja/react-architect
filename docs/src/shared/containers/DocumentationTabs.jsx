import React, { PureComponent, PropTypes } from 'react';
import withRouter from 'react-router/lib/withRouter';
// import { connect } from 'react-redux';
import { Tabs, Tab } from 'react-md/lib/Tabs';

@withRouter
export default class DocumentationTabs extends PureComponent {
  static propTypes = {
    router: PropTypes.shape({
      replace: PropTypes.func.isRequired,
      location: PropTypes.shape({
        pathname: PropTypes.string.isRequired,
        query: PropTypes.shape({
          tab: PropTypes.string,
        }),
      }).isRequired,
      params: PropTypes.shape({
        component: PropTypes.string,
        section: PropTypes.string,
      }).isRequired,
    }).isRequired,
    className: PropTypes.string,
    pathname: PropTypes.string.isRequired,
    children: PropTypes.node,
  };

  constructor(props) {
    super(props);

    this.state = {};
    this._handleTabChange = this._handleTabChange.bind(this);
  }

  _handleTabChange(activeTabIndex) {
    const { router: { replace, location: { pathname } } } = this.props;
    let query;
    if (activeTabIndex > 0) {
      query = { tab: activeTabIndex };
    }

    replace({ pathname, query });
  }

  render() {
    const {
      router: {
        location: { query: { tab } },
        params: { section },
      },
      pathname,
    } = this.props;
    const colors = pathname.indexOf('colors') !== -1;
    const themes = pathname.indexOf('themes') !== -1;
    const customization = pathname.match(/customization/);
    const firstTabLabel = customization && !pathname.match(/grid/) ? 'Info' : 'Examples';

    let propTypesTab;
    if (!customization) {
      propTypesTab = <Tab label="Prop Types" id="documentation-prop-types" />;
    }

    let sassdocTab;
    if (section !== 'helpers' || colors || pathname.indexOf('layovers') !== -1) {
      sassdocTab = (
        <Tab label="SassDoc" id="documentation-sassdoc" />
      );
    }

    let themer;
    if (themes) {
      themer = <Tab label="Theme Builder" id="theme-builder" />;
    }

    return (
      <Tabs
        tabId="documentation"
        className="documentation-tabs"
        activeTabIndex={parseInt(tab, 10) || 0}
        onTabChange={this._handleTabChange}
        colored
      >
        <Tab label={firstTabLabel} id={`documentation-${firstTabLabel.toLowerCase()}`} />
        {propTypesTab}
        {themer}
        {sassdocTab}
      </Tabs>
    );
  }
}
