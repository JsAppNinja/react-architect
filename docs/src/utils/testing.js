/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { StaticRouter } from 'react-router';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { render, shallow, mount } from 'enzyme';
import rootReducer from 'state';

/**
 * I'm lazy.
 *
 * Could be lazier and do `expectSnapshot` though..
 */
export function createSnapshot(children) {
  return renderer.create(children).toJSON();
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * uses something from react-router.
 */
export function createRouterSnapshot(children, location = '/', context = {}) {
  return createSnapshot(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * requires the redux provider.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function createReduxSnapshot(children, state) {
  const store = createStore(rootReducer, state);
  return createSnapshot(
    <Provider store={store}>
      {children}
    </Provider>
  );
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * requires both the redux Provider and something from react-router.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function createReduxRouterSnapshot(children, state, location = '/', context = {}) {
  const store = createStore(rootReducer, state);
  return createSnapshot(
    <Provider store={store}>
      <StaticRouter location={location} context={context}>
        {children}
      </StaticRouter>
    </Provider>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component relies on react-router. This should be used when a component
 * or child component uses findDOMNode or the CSSTransitionGroup, otherwise the createRouterSnapshot should
 * be used.
 */
export function renderRouterSnapshot(children, location = '/', context = {}) {
  return render(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component requires the redux Provider. This should be used when a component
 * or child component uses findDOMNode or the CSSTransitionGroup, otherwise the createReduxSnapshot should
 * be used.
 */
export function renderReduxSnapshot(children, state) {
  const store = createStore(rootReducer, state);
  return render(
    <Provider store={store}>
      {children}
    </Provider>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component requires both the redux Provider and something from react-router.
 * This should be used when a component * or child component uses findDOMNode or the CSSTransitionGroup,
 * otherwise the createReduxRouterSnapshot should be used.
 */
export function renderReduxRouterSnapshot(children, state, location = '/', context = {}) {
  const store = createStore(rootReducer, state);
  return render(
    <Provider store={store}>
      <StaticRouter location={location} context={context}>
        {children}
      </StaticRouter>
    </Provider>
  );
}

/**
 * A simple wrapper to use enzyme's shallow rendering when the component or a child
 * requires something from react-router.
 */
export function shallowWithRouter(children, location = '/', context = {}) {
  return shallow(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}


/**
 * A simple wrapper to use enzyme's mount rendering when the component or a child
 * requires something from react-router.
 */
export function mountWithRouter(children, location = '/', context = {}) {
  return mount(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * A simple wrapper to use enzyme's shallow rendering when the component or a child
 * requires something from react-redux.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function shallowWithProvider(children, state) {
  const store = createStore(rootReducer, state);
  return shallow(<Provider store={store}>{children}</Provider>);
}

/**
 * A simple wrapper to use enzyme's mount rendering when the component or a child
 * requires something from react-redux.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function mountWithProvider(children, state) {
  const store = createStore(rootReducer, state);
  return mount(<Provider store={store}>{children}</Provider>);
}

/**
 * A utility function that will override the console's behavior and throw errors
 * if console.error occurs. This is useful when you want to consider React warnings
 * as failures.
 */
export function captureConsole(match = null) {
  const console = global.console;

  beforeAll(() => {
    global.console = {
      ...console,
      error: (error) => {
        if (match === null || error.match(match)) {
          throw new Error(error);
        }
      },
    };
  });

  afterAll(() => {
    global.console = console;
  });
}
