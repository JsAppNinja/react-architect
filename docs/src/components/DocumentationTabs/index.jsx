import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import Tabs from 'react-md/lib/Tabs/Tabs';
import Tab from 'react-md/lib/Tabs/Tab';
import { stringify } from 'qs';
import { getTab } from 'utils/routing';

import './_styles.scss';

@withRouter
export default class DocumentationTabs extends PureComponent {
  static propTypes = {
    visible: PropTypes.bool,
    location: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
  };

  handleTabChange = (activeTabIndex) => {
    const { history, location: { pathname } } = this.props;
    let search;
    if (activeTabIndex > 0) {
      search = stringify({ tab: activeTabIndex });
    }

    history.replace({ pathname, search });
  };

  render() {
    const { location: { pathname, search }, visible } = this.props;
    if (!visible) {
      return null;
    }

    const activeTabIndex = getTab(search) || 0;
    const colors = pathname.indexOf('colors') !== -1;
    const customization = pathname.indexOf('customization') !== -1;
    const firstTabLabel = customization && pathname.indexOf('grid') === -1
      ? 'Info' : 'Examples';

    let themer;
    let sassdocTab;
    let propTypesTab;
    if (!customization) {
      propTypesTab = <Tab label="Prop Types" id="documentation-prop-types" key="prop-types" />;
    }

    if (colors || pathname.match(/layovers/) || !pathname.match(/helpers|svg/)) {
      sassdocTab = <Tab label="SassDoc" id="documentation-sassdoc" key="sassdoc" />;
    }

    return (
      <Tabs
        mobile={false}
        tabId="documentation"
        className="documentation-tabs"
        activeTabIndex={activeTabIndex}
        onTabChange={this.handleTabChange}
      >
        <Tab label={firstTabLabel} id={`documentation-${firstTabLabel.toLowerCase()}`} key="first-tab" />
        {propTypesTab}
        {themer}
        {sassdocTab}
      </Tabs>
    );
  }
}
