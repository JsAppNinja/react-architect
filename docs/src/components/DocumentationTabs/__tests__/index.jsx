/* eslint-env jest */
import React from 'react';
import { renderRouterSnapshot, mountWithRouter } from 'utils/testing';

import DocumentationTabs from '../';

describe('DocumentationTabs', () => {
  it('should render as null when not visible', () => {
    const tree = renderRouterSnapshot(<DocumentationTabs visible={false} />);
    expect(tree).toMatchSnapshot();
  });

  it('should correctly render on customization routes', () => {
    const tree1 = renderRouterSnapshot(<DocumentationTabs visible />, '/customization/colors');
    const tree2 = renderRouterSnapshot(<DocumentationTabs visible />, '/customization/grids');
    const tree3 = renderRouterSnapshot(<DocumentationTabs visible />, '/customization/themes');
    expect(tree1).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
    expect(tree3).toMatchSnapshot();
  });

  it('should render correctly on component routes', () => {
    const tree1 = renderRouterSnapshot(<DocumentationTabs visible />, '/components/autocompletes');
    const tree2 = renderRouterSnapshot(<DocumentationTabs visible />, '/components/helpers/layovers');
    const tree3 = renderRouterSnapshot(<DocumentationTabs visible />, '/components/helpers/portals');
    expect(tree1).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
    expect(tree3).toMatchSnapshot();
  });

  it('should add the Prop Types tab when not on customization routes', () => {
    const tabs = mountWithRouter(<DocumentationTabs visible />, '/components/autocompletes');
    expect(tabs.find('#documentation-prop-types').length).toBeGreaterThan(0);
  });

  it('should add the Sass Doc tab on the correct routes', () => {
    const tabs1 = mountWithRouter(<DocumentationTabs visible />, '/components/autocompletes');
    const tabs2 = mountWithRouter(<DocumentationTabs visible />, '/components/helpers/portals');
    const tabs3 = mountWithRouter(<DocumentationTabs visible />, '/components/helpers/layovers');
    const tabs4 = mountWithRouter(<DocumentationTabs visible />, '/customization/colors');

    expect(tabs1.find('#documentation-sassdoc').length).toBeGreaterThan(0);
    expect(tabs2.find('#documentation-sassdoc').length).toBe(0);
    expect(tabs3.find('#documentation-sassdoc').length).toBeGreaterThan(0);
    expect(tabs4.find('#documentation-sassdoc').length).toBeGreaterThan(0);
  });
});
