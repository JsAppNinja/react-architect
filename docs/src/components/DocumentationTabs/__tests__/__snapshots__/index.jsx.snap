// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentationTabs should correctly render on customization routes 1`] = `
<ul
  class="md-tabs documentation-tabs"
  role="tablist"
>
  <span
    aria-hidden="true"
  />
  <li
    aria-controls="documentation-panel-0"
    aria-pressed="false"
    aria-selected="true"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--active"
    id="documentation-info"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Info
    </div>
  </li>
  <li
    aria-controls="documentation-panel-1"
    aria-pressed="false"
    aria-selected="false"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="documentation-sassdoc"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      SassDoc
    </div>
  </li>
  <span
    class="md-tab-indicator"
    style="height:2px;width:160px;-webkit-transform:translate3d(0px, 0, 0);-moz-transform:translate3d(0px, 0, 0);-ms-transform:translate3d(0px, 0, 0);transform:translate3d(0px, 0, 0);"
  />
</ul>
`;

exports[`DocumentationTabs should correctly render on customization routes 2`] = `
<ul
  class="md-tabs documentation-tabs"
  role="tablist"
>
  <span
    aria-hidden="true"
  />
  <li
    aria-controls="documentation-panel-0"
    aria-pressed="false"
    aria-selected="true"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--active"
    id="documentation-examples"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Examples
    </div>
  </li>
  <li
    aria-controls="documentation-panel-1"
    aria-pressed="false"
    aria-selected="false"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="documentation-sassdoc"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      SassDoc
    </div>
  </li>
  <span
    class="md-tab-indicator"
    style="height:2px;width:160px;-webkit-transform:translate3d(0px, 0, 0);-moz-transform:translate3d(0px, 0, 0);-ms-transform:translate3d(0px, 0, 0);transform:translate3d(0px, 0, 0);"
  />
</ul>
`;

exports[`DocumentationTabs should correctly render on customization routes 3`] = `
<ul
  class="md-tabs documentation-tabs"
  role="tablist"
>
  <span
    aria-hidden="true"
  />
  <li
    aria-controls="documentation-panel-0"
    aria-pressed="false"
    aria-selected="true"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--active"
    id="documentation-info"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Info
    </div>
  </li>
  <li
    aria-controls="documentation-panel-1"
    aria-pressed="false"
    aria-selected="false"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="documentation-sassdoc"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      SassDoc
    </div>
  </li>
  <span
    class="md-tab-indicator"
    style="height:2px;width:160px;-webkit-transform:translate3d(0px, 0, 0);-moz-transform:translate3d(0px, 0, 0);-ms-transform:translate3d(0px, 0, 0);transform:translate3d(0px, 0, 0);"
  />
</ul>
`;

exports[`DocumentationTabs should render as null when not visible 1`] = `null`;

exports[`DocumentationTabs should render correctly on component routes 1`] = `
<ul
  class="md-tabs documentation-tabs"
  role="tablist"
>
  <span
    aria-hidden="true"
  />
  <li
    aria-controls="documentation-panel-0"
    aria-pressed="false"
    aria-selected="true"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--active"
    id="documentation-examples"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Examples
    </div>
  </li>
  <li
    aria-controls="documentation-panel-1"
    aria-pressed="false"
    aria-selected="false"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="documentation-prop-types"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Prop Types
    </div>
  </li>
  <li
    aria-controls="documentation-panel-2"
    aria-pressed="false"
    aria-selected="false"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="documentation-sassdoc"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      SassDoc
    </div>
  </li>
  <span
    class="md-tab-indicator"
    style="height:2px;width:160px;-webkit-transform:translate3d(0px, 0, 0);-moz-transform:translate3d(0px, 0, 0);-ms-transform:translate3d(0px, 0, 0);transform:translate3d(0px, 0, 0);"
  />
</ul>
`;

exports[`DocumentationTabs should render correctly on component routes 2`] = `
<ul
  class="md-tabs documentation-tabs"
  role="tablist"
>
  <span
    aria-hidden="true"
  />
  <li
    aria-controls="documentation-panel-0"
    aria-pressed="false"
    aria-selected="true"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--active"
    id="documentation-examples"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Examples
    </div>
  </li>
  <li
    aria-controls="documentation-panel-1"
    aria-pressed="false"
    aria-selected="false"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="documentation-prop-types"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Prop Types
    </div>
  </li>
  <li
    aria-controls="documentation-panel-2"
    aria-pressed="false"
    aria-selected="false"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="documentation-sassdoc"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      SassDoc
    </div>
  </li>
  <span
    class="md-tab-indicator"
    style="height:2px;width:160px;-webkit-transform:translate3d(0px, 0, 0);-moz-transform:translate3d(0px, 0, 0);-ms-transform:translate3d(0px, 0, 0);transform:translate3d(0px, 0, 0);"
  />
</ul>
`;

exports[`DocumentationTabs should render correctly on component routes 3`] = `
<ul
  class="md-tabs documentation-tabs"
  role="tablist"
>
  <span
    aria-hidden="true"
  />
  <li
    aria-controls="documentation-panel-0"
    aria-pressed="false"
    aria-selected="true"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--active"
    id="documentation-examples"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Examples
    </div>
  </li>
  <li
    aria-controls="documentation-panel-1"
    aria-pressed="false"
    aria-selected="false"
    class="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="documentation-prop-types"
    role="tab"
    tabindex="0"
  >
    <div
      class="md-tab-label"
    >
      Prop Types
    </div>
  </li>
  <span
    class="md-tab-indicator"
    style="height:2px;width:160px;-webkit-transform:translate3d(0px, 0, 0);-moz-transform:translate3d(0px, 0, 0);-ms-transform:translate3d(0px, 0, 0);transform:translate3d(0px, 0, 0);"
  />
</ul>
`;
